test/jdk/tools/jar/modularJar/Basic.java

```diff
@@ -38,6 +40,8 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import jdk.internal.module.ModuleReferenceImpl;
+import jdk.internal.module.ModuleResolution;
```

Comment:
At some point we need to put in test infrastructure to avoid this and a few other tests from depending on JDK internals.

Reply:
Do you think it is better to parse the output of `javap` instead of using the internals to detect if it worked?

Reply:
Parsing the output of javap would be fragile. Instead, I think we'll just add some test infrastructure at some point, maybe when this class file attribute is documented somewhere.